apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/CoreConfiguration.java

```diff
@@ -166,7 +167,7 @@
             "You must use the query bar to filter for a specific environment in versions prior to 7.2.")
         .build();
 
-    private final ConfigurationOption<Double> sampleRate = RoundedDoubleConverter.withPrecision(4)
+    private final ConfigurationOption<Double> sampleRate = RoundedDoubleConverter.withPrecision(SAMPLE_RATE_PRECISION_DIGITS)
```

Comment:
[minor] since the entire logic now resides within `RoundedDoubleConverter`, there is no need for this constant.

Reply:
Good point, I've added a public static factory method to hide this, we always have ways to reuse with higher/lower precision if required (was already tested thus no harm to keep it).